### PR TITLE
Temporal docs added to expressions.rst

### PR DESCRIPTION
**Pull request by sunaysanghani**
@jaychia @colin-ho Just added temporal doc section to expressions.rst. Let me know what you think of the content and then we can finalize which page or user-guide section to put it on from there. Thanks!

### CODE REVIEW

File: docs/source/user_guide/basic_concepts/expressions.rst [hunk 1]
```diff
@@ -314,3 +314,188 @@ The :meth:`.if_else() <daft.expressions.Expression.if_else>` method is a useful
...
+    │ {"a": 3, "b": 4} ┆ 3    │
+    ╰──────────────────┴──────╯
+
+    (Showing first 2 of 2 rows)
+
+
+Using the Truncate Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `truncate` function can be used to truncate dates to a specific time unit. This can be useful when you want to group or round dates to the nearest week, month, etc.
```
Comment (colin-ho) [inline docs/source/user_guide/basic_concepts/expressions.rst:459]:
```suggestion
The `truncate` function can be used to truncate timestamps to a specific time unit. For example, you can use it to truncate timestamps to the nearest day.
```

File: docs/source/user_guide/basic_concepts/expressions.rst [hunk 2]
```diff
@@ -314,3 +314,188 @@ The :meth:`.if_else() <daft.expressions.Expression.if_else>` method is a useful
...
+    import pandas as pd
+    import daft
+
+    # Create a DataFrame with a range of dates
+    df = daft.from_pydict({
+        "dates": pd.date_range(start="2023-07-01", end="2023-07-06", freq='D').to_list()
+    })
+
+    # Truncate dates to the start of the week
+    df = df.with_column("truncated_dates", df["dates"].dt.truncate(freq="W-SUN"))
```
Comment (colin-ho) [inline docs/source/user_guide/basic_concepts/expressions.rst:472]: Does this code work? I'm pretty sure the truncate method doesn't expose a `freq` argument

File: docs/source/user_guide/basic_concepts/expressions.rst [hunk 3]
```diff
@@ -314,3 +314,188 @@ The :meth:`.if_else() <daft.expressions.Expression.if_else>` method is a useful
...
+
+.. code:: python
+
+    df = daft.from_pydict({
+        "json": [
+            '{"a": 1, "b": 2}',
+            '{"a": 3, "b": 4}',
+        ],
+    })
+    df = df.with_column("a", df["json"].json.query(".a"))
```
Comment (colin-ho) [inline docs/source/user_guide/basic_concepts/expressions.rst:438]: Json?

**Comment by sunaysanghani** [comment]
just added revisions, let me know what you think

**Comment by colin-ho** [comment]
> just added revisions, let me know what you think

Nice can you rebase it on main so only the related changes are reflected

**Comment by samster25** [comment]
@colin-ho Status on this PR? Can you clean up and merge?

**Pull request merged**